java/java-psi-impl/src/com/intellij/lang/java/parser/StatementParser.java

```diff
@@ -160,7 +160,14 @@ else if (tokenType == JavaTokenType.IDENTIFIER || tokenType == JavaTokenType.AT)
       skipQualifiedName(builder);
       IElementType suspectedLT = builder.getTokenType(), next = builder.lookAhead(1);
       refPos.rollbackTo();
-      if (suspectedLT == JavaTokenType.LT || suspectedLT == JavaTokenType.DOT && next == JavaTokenType.AT) {
+
+      // skip the the type with possible type parameters to see if it is followed by "::"
+      myParser.getReferenceParser().parseType(builder, 0);
+      final boolean constructorRef = builder.getTokenType() == JavaTokenType.DOUBLE_COLON;
```

Comment:
Statistically, a variable declaration is much more likely here, so it makes sense to check for double colon _after_ that attempt is failed, at line 168 (in original code).

Reply:
@trespasserw, Thank you for the review.
I moved the check if the examined line is a method reference to a constructor to the `if`-statement below as you suggested